doc/BUILD.rst

```diff
@@ -0,0 +1,500 @@
+advice. (For example, it can detect potential null-pointer exceptions that may
+not be obvious to the human eye.)
+
+.. code:: sh
+
+   meson configure build/ -Dwerror=true
+
+Deprecation Warnings (``-Ddeprecation_warnings``)
+"""""""""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``(null)``
+* Options: ``error``, ``no-error``, ``ignore``
+
+Sets deprecation warnings to either hard-fail (``error``), print as warnings but
+not trigger full errors if ``-Dwerror=true`` (``no-error``), and otherwise
+ignore them (``ignore``).
+
+Generally, ``no-error`` is the recommended default when using ``-Dwerror=true``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeprecation_warnings=no-error
+
+
+In-Game Developer Options (``-Ddeveloper``)
+"""""""""""""""""""""""""""""""""""""""""""
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+For testing actual gameplay, you can set this option and it will enable cheats
+and other 'fast-forward' options by the pressing keys defined in
+``src/config.h``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeveloper=true
+
+Build Type (``-Dbuildtype``)
+""""""""""""""""""""""""""""
+
+* Default: ``release``
+* Options: ``release``, ``debug``
+
+Sets the type of build. ``debug`` enables several additional debugging features.
+Useful for development.
+
+.. code:: sh
+
+   meson configure build/ -Dbuildtype=debug
+
+Debugging (``-Db_ndebug``)
+""""""""""""""""""""""""""
+
+* Default: ``true``
+* Options: ``true``, ``false``
+
+The name of this flag is opposite of what you'd expect. Think of it as "Not
+Debugging". If ``true``, it is *not* a debug build. If ``false``, it *is* a
+debugging build.
+
+.. code:: sh
+
+   meson configure build/ -Db_ndebug=false
+
+Stack Trace Debugging (``-Db_sanitize``)
+""""""""""""""""""""""""""""""""""""""""
+
+This is useful for debugging crashes in the game. It uses
+`AddressSanitizer <https://github.com/google/sanitizers/wiki/AddressSanitizer>`__:
+
+.. code:: sh
+
+   meson configure build/ -Db_sanitize=address,undefined
+
+Depending on your platform, you may need to specify the specific library binary
+to use to launch ASan appropriately. Using macOS as an example:
+
+.. code:: sh
+
+   export DYLD_INSERT_LIBRARIES=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.0/lib/darwin/libclang_rt.asan_osx_dynamic.dylib
+
+The ``../12.0.0/..`` in the path of ``DYLD_INSERT_LIBRARIES`` changes with each
+version of Xcode. If it fails to launch for you, ensure that the version number
+is correct by browsing to the parent directory of ``../clang``.
+
+Then, you can launch Taisei's binary from the command line (using macOS as an
+example):
+
+.. code:: sh
+
+   /path/to/Taisei.app/Contents/MacOS/Taisei
+
+Linking & Striping (``-Db_lto``/``-Dstrip``)
+""""""""""""""""""""""""""""""""""""""""""""
+
+* Defaults: ``false``
+* Options: ``true``, ``false``
+
+These options prevent stripping of the binaries, leading to faster build times
```

Comment:
`b_lto` controls link-time optimizations. They increase build times by a lot and increases runtime performance. 
`strip=true` removes debug symbols. I never noticed that it increases build times… does it? I don’t think it makes a difference for the runtime performance. It increases binary size though.